---
title: '2018 annual review: Microbial communities performing dissimilatory nitrate
  reduction to ammonium in Australian forest and agricultural soils'
author: "Eric Ireland"
date: "16 March 2018"
output:
  html_document: default
  word_document: default
bibliography: 2018-03-13_bib.bib
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(lattice)
library(readr)
library(ggplot2)
library(scales)
library(car)
narG <- read_csv("narG.csv")
nirS <- read_csv("nirS.csv")
AOA_amoA <- read_csv("AOA_amoA.csv")
nosZ <- read_csv("nosZ.csv")
rRNA16S <- read_csv("rRNA16S.csv")
nirK <- read_csv("nirK.csv")
```


## Chapter 1: N functional gene abundance and gross N transformation rates

Soil was collected on from an area of cool temperate rainforest in Yarra State Forest near Powelltown in Victoria, Australia. Three sampling sites of 10 m × 10 m were selected; each had a similar soil texture (sandy loam). To ensure sampling locations with different vegetation and riparian and terrestrial zones were represented, a strict randomisation of sampling locations was not possible for this experiment.

Soil was collected from a wet gully with dominant vegetation consisting of *Nothofagus cunninghami* (Myrtle Beech), a nearby slope with a mix of *N. cunninghami* and Eucalyptus regnans (Mountain Ash), and a slope with *Eucalyptus regnans* (Mountain Ash) predominating. Some surface undecomposed organic matter (O1 horizon) was removed, and approximately 5 kg soil was collected from a depth of 0 cm to 10 cm in the A-horizon.
Soil was stored on ice for 2 h during transportation to the laboratory, where they were stored at 4&deg;C until the beginning of the incubation. Soils were combined and partially air-dried. The bulk density of the sieved soil was determined by drying three subsamples of a known volume at 105&deg;C for 72 h and then measuring their weights [@blake_bulk_1986]. Approximately 80 g fresh soil from each site was frozen at −80&deg;C and freeze dried for long term storage before DNA analysis. The remaining soil from all sites was then air-dried, combined, mixed thoroughly, sieved to 2 mm and stored at 4&deg;C for use in the incubation experiments.

An incubation experiment was used to determine the effects of different levels of N addition on abundance of genes involved in DNRA, denitri cation and nitrification. N was added in the form of NH~4~NO~3~ at a rate of 25 μg.g^−1^, 75 μg.g^−1^ and 150 μg.g^−1^ soil. Soil samples of 10 g were placed in 250 ml plastic vials and incubated at 25&deg;C and 60 % WFPS. Each sample had three replicates. Soils were destructively sampled after 15 min (for day 0 measurements), then at 7 d, 14 d and 28 d. The 16S rRNA gene, DNRA gene (*nrfA*) denitrification (*nirK*,*nirS*,*narG*,*nosZ*) and nitrification (AOA *amoA* and AOB *amoA*) genes were quantified using real-time PCR (qPCR).

### Results

#### *nrfA* abundance

I have cloned *nrfA* using two different primer sets. I have decided to use the primers developed by @welsh_refined_2014, because these were published most recently, and should amplify the widest range of sequences. So far, the qPCR efficiency has been either below 60%, or above 140%, depending on the PCR conditions used. A possible reason for this problem could be something inhibiting the reaction, so I extracted fresh plasmids and have continued to experiment with different protocols to try and get efficiency > 80%. I have managed to improve the efficiency marginally by lower the annealing temperature to 50&deg;C, however this approach could lower the PCR specificity. Although these primers have been used for qPCR in some recent studies, some other researchers have been unable to achieve qPCR efficiency above 50% for *nrfA*, e.g. @jones_sulfide-induced_2017.

To try and achieve greater efficiency in qPCR for *nrfA* and other N cycling genes I will try the following additional strategies:

* Linearize the plasmids using appropriate restriction enzymes prior to performing qPCR [@chen_dna_2007;@hou_serious_2010].
* Try different mastermixes. Compare different mastermixes using the same plasmids and primers. 
* Clone the genes again from new soil samples.
* Clone the genes from *Escherichia coli* or other pure cultures instead of from soil DNA.

#### Bacterial 16S rRNA abundance


```{r rRNA16S, echo=FALSE, fig.cap="Results of qPCR for *rRNA16S* gene. Error bars show SEM. Efficiency = 80.7%, $R^{2} = 0.99$. \\label{rRNA16Sgraph}"}
rRNA16S$copiespergram <- (rRNA16S$SQ*1000)/0.25 #100ÂµL of solution is recovered from 0.25g soil. This template DNA is diluted 1:10 and then 1ÂµL of that diluted DNA is used as template. So we're counting copies of DNA in 0.1ÂµL of template. So, we are using 0.001 (1/100th) of the original DNA. So, times by 100 to get the amount in 100ÂµL DNA, then times by 4 (i.e. divide by 0.25) to get the copies in 1g of soil.
rRNA16S$log10copiespergram <- log10((rRNA16S$SQ*1000)/0.25) #log base 10 transform data
rRNA16S$logecopiespergram <- log((rRNA16S$SQ*1000)/0.25) #log base e transform data

colnames(rRNA16S)[3] <- "time" # renamed day column to time



### works!!

#summarySE function for errorbars from http://www.cookbook-r.com/Graphs/Plotting_means_and_error_bars_(ggplot2)/#Helper%20functions
summarySE <- function(data=NULL, measurevar, groupvars=NULL, na.rm=FALSE,
                      conf.interval=.95, .drop=TRUE) {
  library(plyr)
  
  # New version of length which can handle NA's: if na.rm==T, don't count them
  length2 <- function (x, na.rm=FALSE) {
    if (na.rm) sum(!is.na(x))
    else       length(x)
  }
  
  # This does the summary. For each group's data frame, return a vector with
  # N, mean, and sd
  datac <- ddply(data, groupvars, .drop=.drop,
                 .fun = function(xx, col) {
                   c(N    = length2(xx[[col]], na.rm=na.rm),
                     mean = mean   (xx[[col]], na.rm=na.rm),
                     sd   = sd     (xx[[col]], na.rm=na.rm)
                   )
                 },
                 measurevar
  )
  
  # Rename the "mean" column    
  datac <- rename(datac, c("mean" = measurevar))
  
  datac$se <- datac$sd / sqrt(datac$N)  # Calculate standard error of the mean
  
  # Confidence interval multiplier for standard error
  # Calculate t-statistic for confidence interval: 
  # e.g., if conf.interval is .95, use .975 (above/below), and use df=N-1
  ciMult <- qt(conf.interval/2 + .5, datac$N-1)
  datac$ci <- datac$se * ciMult
  
  return(datac)
}

rRNA16S_summary <- summarySE(rRNA16S, measurevar="logecopiespergram", groupvars=c("time","treatment")) #not sure if this is correct!

rRNA16S2 <- rRNA16S_summary
rRNA16S2$treatment <- factor(rRNA16S2$treatment, levels=c("C","25","75","150"))
rRNA16S2$time <- factor(rRNA16S2$time)


# Error bars represent standard error of the mean
ggplot(rRNA16S2, aes(x=time, y=logecopiespergram, fill=treatment)) + 
    geom_bar(position=position_dodge(), stat="identity") +
    geom_errorbar(aes(ymin=logecopiespergram-se, ymax=logecopiespergram+se),
                  width=.2,                    # Width of the error bars
                  position=position_dodge(.9))+
  labs(
    x = "time (days)",
    y = "log(copies per gram soil)",
    colour = "treatment"
  )
```

```{r echo=FALSE}
linear.model.rRNA <- lm(formula = logecopiespergram ~ treatment + time, data = rRNA16S)
anovatablerRNA <- Anova(linear.model.rRNA, type="II")
rRNAp <- unname(anovatablerRNA$`Pr(>F)`[1])
```

<!-- could use `r round(rRNAp,3)` to report p value. -->
16S rRNA gene abundance did not differ between treatments (*F*-test, $p=`r ifelse(round(rRNAp, 3) == 0, "<.001", round(rRNAp, 3))`$).


#### AOA *amoA* abundance

```{r AOA_amoA, echo=FALSE, fig.cap="Results of qPCR for AOA *amoA* gene. Error bars show SEM. Efficiency = 93.4%, $R^{2} = 0.97$.\\label{AOA_amoAgraph}"}
AOA_amoA$copiespergram <- (AOA_amoA$SQ*1000)/0.25 #100ÂµL of solution is recovered from 0.25g soil. This template DNA is diluted 1:10 and then 1ÂµL of that diluted DNA is used as template. So we're counting copies of DNA in 0.1ÂµL of template. So, we are using 0.001 (1/1000th) of the original DNA. So, times by 1000 to get the amount in 100ÂµL DNA, then times by 4 (i.e. divide by 0.25) to get the copies in 1g of soil.
AOA_amoA$log10copiespergram <- log10((AOA_amoA$SQ*1000)/0.25) #log base 10 transform data
AOA_amoA$logecopiespergram <- log((AOA_amoA$SQ*1000)/0.25) #log base e transform data

colnames(AOA_amoA)[3] <- "time" # renamed day column to time



### works!!

#summarySE function for errorbars from http://www.cookbook-r.com/Graphs/Plotting_means_and_error_bars_(ggplot2)/#Helper%20functions
summarySE <- function(data=NULL, measurevar, groupvars=NULL, na.rm=FALSE,
                      conf.interval=.95, .drop=TRUE) {
  library(plyr)
  
  # New version of length which can handle NA's: if na.rm==T, don't count them
  length2 <- function (x, na.rm=FALSE) {
    if (na.rm) sum(!is.na(x))
    else       length(x)
  }
  
  # This does the summary. For each group's data frame, return a vector with
  # N, mean, and sd
  datac <- ddply(data, groupvars, .drop=.drop,
                 .fun = function(xx, col) {
                   c(N    = length2(xx[[col]], na.rm=na.rm),
                     mean = mean   (xx[[col]], na.rm=na.rm),
                     sd   = sd     (xx[[col]], na.rm=na.rm)
                   )
                 },
                 measurevar
  )
  
  # Rename the "mean" column    
  datac <- rename(datac, c("mean" = measurevar))
  
  datac$se <- datac$sd / sqrt(datac$N)  # Calculate standard error of the mean
  
  # Confidence interval multiplier for standard error
  # Calculate t-statistic for confidence interval: 
  # e.g., if conf.interval is .95, use .975 (above/below), and use df=N-1
  ciMult <- qt(conf.interval/2 + .5, datac$N-1)
  datac$ci <- datac$se * ciMult
  
  return(datac)
}

AOA_amoA_summary <- summarySE(AOA_amoA, measurevar="logecopiespergram", groupvars=c("time","treatment")) #not sure if this is correct!

AOA_amoA2 <- AOA_amoA_summary
AOA_amoA2$treatment <- factor(AOA_amoA2$treatment, levels=c("C","25","75","150"))
AOA_amoA2$time <- factor(AOA_amoA2$time)

# Use 95% confidence intervals instead of SEM
ggplot(AOA_amoA2, aes(x=time, y=logecopiespergram, fill=treatment)) + 
  geom_bar(position=position_dodge(), stat="identity") +
  geom_errorbar(aes(ymin=logecopiespergram-se, ymax=logecopiespergram+se),
                width=.2,                    # Width of the error bars
                position=position_dodge(.9))+
  labs(
    x = "time (days)",
    y = "log(copies per gram soil)",
    colour = "treatment"
  )

       
```

```{r echo=FALSE}
linear.model.AOA_amoA <- lm(formula = logecopiespergram ~ treatment + time, data = AOA_amoA)
anovatable_AOA_amoA <- Anova(linear.model.AOA_amoA, type="II")
AOA_amoA_p <- unname(anovatable_AOA_amoA$`Pr(>F)`[1])
```

<!-- could use `r round(rRNAp,3)` to report p value. -->
AOA *amoA* gene abundance did not differ between treatments (*F*-test, $p=`r ifelse(AOA_amoA_p< 0.05, "<0.05", round(AOA_amoA_p, 3))`$).

#### *nosZ* abundance

```{r nosZ, echo=FALSE, fig.cap="Results of qPCR for *nosZ* gene. Error bars show SEM. Efficiency = 97.0%, $R^{2} = 0.99$.\\label{nosZgraph}"}
nosZ$copiespergram <- (nosZ$SQ*1000)/0.25 #100ÂµL of solution is recovered from 0.25g soil. This template DNA is diluted 1:10 and then 1ÂµL of that diluted DNA is used as template. So we're counting copies of DNA in 0.1ÂµL of template. So, we are using 0.001 (1/1000th) of the original DNA. So, times by 1000 to get the amount in 100ÂµL DNA, then times by 4 (i.e. divide by 0.25) to get the copies in 1g of soil.
nosZ$log10copiespergram <- log10((nosZ$SQ*1000)/0.25) #log base 10 transform data
nosZ$logecopiespergram <- log((nosZ$SQ*1000)/0.25) #log base e transform data

colnames(nosZ)[3] <- "time" # renamed day column to time



### works!!

#summarySE function for errorbars from http://www.cookbook-r.com/Graphs/Plotting_means_and_error_bars_(ggplot2)/#Helper%20functions
summarySE <- function(data=NULL, measurevar, groupvars=NULL, na.rm=FALSE,
                      conf.interval=.95, .drop=TRUE) {
  library(plyr)
  
  # New version of length which can handle NA's: if na.rm==T, don't count them
  length2 <- function (x, na.rm=FALSE) {
    if (na.rm) sum(!is.na(x))
    else       length(x)
  }
  
  # This does the summary. For each group's data frame, return a vector with
  # N, mean, and sd
  datac <- ddply(data, groupvars, .drop=.drop,
                 .fun = function(xx, col) {
                   c(N    = length2(xx[[col]], na.rm=na.rm),
                     mean = mean   (xx[[col]], na.rm=na.rm),
                     sd   = sd     (xx[[col]], na.rm=na.rm)
                   )
                 },
                 measurevar
  )
  
  # Rename the "mean" column    
  datac <- rename(datac, c("mean" = measurevar))
  
  datac$se <- datac$sd / sqrt(datac$N)  # Calculate standard error of the mean
  
  # Confidence interval multiplier for standard error
  # Calculate t-statistic for confidence interval: 
  # e.g., if conf.interval is .95, use .975 (above/below), and use df=N-1
  ciMult <- qt(conf.interval/2 + .5, datac$N-1)
  datac$ci <- datac$se * ciMult
  
  return(datac)
}

nosZ_summary <- summarySE(nosZ, measurevar="logecopiespergram", groupvars=c("time","treatment")) #not sure if this is correct!

nosZ2 <- nosZ_summary
nosZ2$treatment <- factor(nosZ2$treatment, levels=c("C","25","75","150"))
nosZ2$time <- factor(nosZ2$time)

# Use 95% confidence intervals instead of SEM
ggplot(nosZ2, aes(x=time, y=logecopiespergram, fill=treatment)) + 
  geom_bar(position=position_dodge(), stat="identity") +
  geom_errorbar(aes(ymin=logecopiespergram-se, ymax=logecopiespergram+se),
                width=.2,                    # Width of the error bars
                position=position_dodge(.9))+
  labs(
    x = "time (days)",
    y = "log(copies per gram soil)",
    colour = "treatment"
  )

       
```

```{r echo=FALSE}
linear.model.nosZ <- lm(formula = logecopiespergram ~ treatment + time, data = nosZ)
anovatable_nosZ <- Anova(linear.model.nosZ, type="II")
nosZ_p <- unname(anovatable_nosZ$`Pr(>F)`[1])
```

<!-- could use `r round(rRNAp,3)` to report p value. -->
*nosZ* gene abundance varied between treatments, increasing and decreasing the relative gene abundance at different times with increased NH~4~NO~3~ addition (*F*-test, $p=`r ifelse(nosZ_p < 0.01, "<0.01", ifelse(nosZ_p < 0.05, "<0.05",round(nosZ_p, 3)))`$).

#### *narG* abundance

```{r narG, echo=FALSE, fig.cap="Results of qPCR for narG gene. Error bars show SEM. Efficiency = 93.1%, $R^{2} = 0.99$\\label{narGgraph}"}
narG$copiespergram <- (narG$SQ*1000)/0.25 #100?L of solution is recovered from 0.25g soil. This template DNA is diluted 1:10 and then 1?L of that diluted DNA is used as template. So we're counting copies of DNA in 0.1?L of template. So, we are using 0.001 (1/1000th) of the original DNA. So, times by 1000 to get the amount in 100?L DNA, then times by 4 (i.e. divide by 0.25) to get the copies in 1g of soil.
narG$log10copiespergram <- log10((narG$SQ*1000)/0.25) #log base 10 transform data
narG$logecopiespergram <- log((narG$SQ*1000)/0.25) #log base e transform data

colnames(narG)[3] <- "time" # renamed day column to time



### works!!

#summarySE function for errorbars from http://www.cookbook-r.com/Graphs/Plotting_means_and_error_bars_(ggplot2)/#Helper%20functions
summarySE <- function(data=NULL, measurevar, groupvars=NULL, na.rm=FALSE,
                      conf.interval=.95, .drop=TRUE) {
  library(plyr)
  
  # New version of length which can handle NA's: if na.rm==T, don't count them
  length2 <- function (x, na.rm=FALSE) {
    if (na.rm) sum(!is.na(x))
    else       length(x)
  }
  
  # This does the summary. For each group's data frame, return a vector with
  # N, mean, and sd
  datac <- ddply(data, groupvars, .drop=.drop,
                 .fun = function(xx, col) {
                   c(N    = length2(xx[[col]], na.rm=na.rm),
                     mean = mean   (xx[[col]], na.rm=na.rm),
                     sd   = sd     (xx[[col]], na.rm=na.rm)
                   )
                 },
                 measurevar
  )
  
  # Rename the "mean" column    
  datac <- rename(datac, c("mean" = measurevar))
  
  datac$se <- datac$sd / sqrt(datac$N)  # Calculate standard error of the mean
  
  # Confidence interval multiplier for standard error
  # Calculate t-statistic for confidence interval: 
  # e.g., if conf.interval is .95, use .975 (above/below), and use df=N-1
  ciMult <- qt(conf.interval/2 + .5, datac$N-1)
  datac$ci <- datac$se * ciMult
  
  return(datac)
}

narGsummary <- summarySE(narG, measurevar="logecopiespergram", groupvars=c("time","treatment")) #not sure if this is correct!

narG2 <- narGsummary
narG2$treatment <- factor(narG2$treatment, levels=c("C","25","75","150"))
narG2$time <- factor(narG2$time)

# Use 95% confidence intervals instead of SEM
ggplot(narG2, aes(x=time, y=logecopiespergram, fill=treatment)) + 
  geom_bar(position=position_dodge(), stat="identity") +
  geom_errorbar(aes(ymin=logecopiespergram-se, ymax=logecopiespergram+se),
                width=.2,                    # Width of the error bars
                position=position_dodge(.9))+
  labs(
    x = "time (days)",
    y = "log(copies per gram soil)",
    colour = "treatment"
  )

       
```

```{r echo=FALSE}
linear.model.narG <- lm(formula = logecopiespergram ~ treatment + time, data = narG)
anovatable_narG <- Anova(linear.model.narG, type="II")
narG_p <- unname(anovatable_narG$`Pr(>F)`[1])
```

*narG* gene abundance did not differ between treatments (*F*-test, $p=`r ifelse(narG_p < 0.01, "<0.01", ifelse(narG_p < 0.05, "<0.05",round(narG_p, 3)))`$).

#### *nirK* abundance

```{r nirK, echo=FALSE, fig.cap="Results of qPCR for *nirK* gene. Error bars show SEM. Efficiency = 84.3%, $R^{2} = 0.99$. These results were from undiluted DNA samples, so I will have to reanalyse the samples using diluted DNA.\\label{nirKgraph}"}
nirK$copiespergram <- (nirK$SQ*1000)/0.25 #100ÂµL of solution is recovered from 0.25g soil. This template DNA is diluted 1:10 and then 1ÂµL of that diluted DNA is used as template. So we're counting copies of DNA in 0.1ÂµL of template. So, we are using 0.001 (1/100th) of the original DNA. So, times by 100 to get the amount in 100ÂµL DNA, then times by 4 (i.e. divide by 0.25) to get the copies in 1g of soil.
nirK$log10copiespergram <- log10((nirK$SQ*1000)/0.25) #log base 10 transform data
nirK$logecopiespergram <- log((nirK$SQ*1000)/0.25) #log base e transform data

colnames(nirK)[3] <- "time" # renamed day column to time



### works!!

#summarySE function for errorbars from http://www.cookbook-r.com/Graphs/Plotting_means_and_error_bars_(ggplot2)/#Helper%20functions
summarySE <- function(data=NULL, measurevar, groupvars=NULL, na.rm=FALSE,
                      conf.interval=.95, .drop=TRUE) {
  library(plyr)
  
  # New version of length which can handle NA's: if na.rm==T, don't count them
  length2 <- function (x, na.rm=FALSE) {
    if (na.rm) sum(!is.na(x))
    else       length(x)
  }
  
  # This does the summary. For each group's data frame, return a vector with
  # N, mean, and sd
  datac <- ddply(data, groupvars, .drop=.drop,
                 .fun = function(xx, col) {
                   c(N    = length2(xx[[col]], na.rm=na.rm),
                     mean = mean   (xx[[col]], na.rm=na.rm),
                     sd   = sd     (xx[[col]], na.rm=na.rm)
                   )
                 },
                 measurevar
  )
  
  # Rename the "mean" column    
  datac <- rename(datac, c("mean" = measurevar))
  
  datac$se <- datac$sd / sqrt(datac$N)  # Calculate standard error of the mean
  
  # Confidence interval multiplier for standard error
  # Calculate t-statistic for confidence interval: 
  # e.g., if conf.interval is .95, use .975 (above/below), and use df=N-1
  ciMult <- qt(conf.interval/2 + .5, datac$N-1)
  datac$ci <- datac$se * ciMult
  
  return(datac)
}

nirK_summary <- summarySE(nirK, measurevar="logecopiespergram", groupvars=c("time","treatment")) #not sure if this is correct!

nirK2 <- nirK_summary
nirK2$treatment <- factor(nirK2$treatment, levels=c("C","25","75","150"))
nirK2$time <- factor(nirK2$time)

# Use 95% confidence intervals instead of SEM
ggplot(nirK2, aes(x=time, y=logecopiespergram, fill=treatment)) + 
  geom_bar(position=position_dodge(), stat="identity") +
  geom_errorbar(aes(ymin=logecopiespergram-se, ymax=logecopiespergram+se),
                width=.2,                    # Width of the error bars
                position=position_dodge(.9))+
  labs(
    x = "time (days)",
    y = "log(copies per gram soil)",
    colour = "treatment"
  )


```

```{r echo=FALSE}
linear.model.nirK <- lm(formula = logecopiespergram ~ treatment + time, data = nirK)
anovatable_nirK <- Anova(linear.model.nirK, type="II")
nirK_p <- unname(anovatable_nirK$`Pr(>F)`[1])
```

*nirK* gene abundance did not differ between treatments (*F*-test, $p=`r ifelse(nirK_p < 0.01, "<0.01", ifelse(nirK_p < 0.05, "<0.05",round(nirK_p, 3)))`$).

#### *nirS* abundance

```{r nirS, echo=FALSE, fig.cap="Results of qPCR for nirS gene. Error bars show SEM. Efficiency = 77.0%, $R^{2} = 0.97$. \\label{nirSgraph}"}
nirS$copiespergram <- (nirS$SQ*1000)/0.25 #100?L of solution is recovered from 0.25g soil. This template DNA is diluted 1:10 and then 1?L of that diluted DNA is used as template. So we're counting copies of DNA in 0.1?L of template. So, we are using 0.001 (1/1000th) of the original DNA. So, times by 1000 to get the amount in 100?L DNA, then times by 4 (i.e. divide by 0.25) to get the copies in 1g of soil.
nirS$log10copiespergram <- log10((nirS$SQ*1000)/0.25) #log base 10 transform data
nirS$logecopiespergram <- log((nirS$SQ*1000)/0.25) #log base e transform data

colnames(nirS)[3] <- "time" # renamed day column to time



### works!!

#summarySE function for errorbars from http://www.cookbook-r.com/Graphs/Plotting_means_and_error_bars_(ggplot2)/#Helper%20functions
summarySE <- function(data=NULL, measurevar, groupvars=NULL, na.rm=FALSE,
                      conf.interval=.95, .drop=TRUE) {
  library(plyr)
  
  # New version of length which can handle NA's: if na.rm==T, don't count them
  length2 <- function (x, na.rm=FALSE) {
    if (na.rm) sum(!is.na(x))
    else       length(x)
  }
  
  # This does the summary. For each group's data frame, return a vector with
  # N, mean, and sd
  datac <- ddply(data, groupvars, .drop=.drop,
                 .fun = function(xx, col) {
                   c(N    = length2(xx[[col]], na.rm=na.rm),
                     mean = mean   (xx[[col]], na.rm=na.rm),
                     sd   = sd     (xx[[col]], na.rm=na.rm)
                   )
                 },
                 measurevar
  )
  
  # Rename the "mean" column    
  datac <- rename(datac, c("mean" = measurevar))
  
  datac$se <- datac$sd / sqrt(datac$N)  # Calculate standard error of the mean
  
  # Confidence interval multiplier for standard error
  # Calculate t-statistic for confidence interval: 
  # e.g., if conf.interval is .95, use .975 (above/below), and use df=N-1
  ciMult <- qt(conf.interval/2 + .5, datac$N-1)
  datac$ci <- datac$se * ciMult
  
  return(datac)
}

nirSsummary <- summarySE(nirS, measurevar="logecopiespergram", groupvars=c("time","treatment")) #not sure if this is correct!

nirS2 <- nirSsummary
nirS2$treatment <- factor(nirS2$treatment, levels=c("C","25","75","150"))
nirS2$time <- factor(nirS2$time)

# Use 95% confidence intervals instead of SEM
ggplot(nirS2, aes(x=time, y=logecopiespergram, fill=treatment)) + 
  geom_bar(position=position_dodge(), stat="identity") +
  geom_errorbar(aes(ymin=logecopiespergram-se, ymax=logecopiespergram+se),
                width=.2,                    # Width of the error bars
                position=position_dodge(.9))+
  labs(
    x = "time (days)",
    y = "log(copies per gram soil)",
    colour = "treatment"
  )

       
```

```{r echo=FALSE}
linear.model.nirS <- lm(formula = logecopiespergram ~ treatment + time, data = nirS)
anovatable_nirS <- Anova(linear.model.nirS, type="II")
nirS_p <- unname(anovatable_nirS$`Pr(>F)`[1])
```

*nirS* gene abundance varied between treatments, tending to reduce the relative gene abundance with increased NH~4~NO~3~ addition (*F*-test, $p=`r ifelse(nirS_p < 0.01, "<0.01", ifelse(nirS_p < 0.05, "<0.05",round(nirS_p, 3)))`$).

#### *nosZ* clade II abundance

I have not been able to achieve efficiency higher than 50% for this gene, but most of my attempts were on the CFX96 qPCR machine, while it was broken. The primers I have been using were pre-diluted (supplied in liquid form). I will retry this gene using fresh primers.

#### *napA* abundance

I was not able to amplify this gene using one set of published primers and PCR conditions and two different PCR enzymes. I used DNA extracted from my soil and five other soil DNA samples. I may try again when using other soil DNA samples.

#### AOB *amoA* abundance

I had a lot of difficulty with this gene; after many attempts, I was still unable to achieve efficiency > 80%, or $R^{2} > 0.98$. Non-specific amplification was present in the negative controls and many of the samples. This was probably caused by primer-dimers, because the melting curve showed two peaks, and multiple bands were present when I ran the PCR products on a gel. I used freshly autoclaved water to try and eliminate any contamination. The soil used for this experiment is very acidic (pH in 1:5 CaCl~2~ = 3.6), so low levels of AOB *amoA* were expected. I could keep working to optimise this assay by titrating a range of primer and MgCl~2~ concentrations, but for the purposes of these samples, I think I can just report that the AOB *amoA* levels were below the detection limit.

### Planned experiments

#### Soil microbial communities and N cycling along an elevational diversity gradient at Mount Baw Baw

SEFS Researchers (Sabine Kasel and Craig Nitschke) at Burnley have weather stations positioned along an elevational gradient at Mount Baw Baw. They have already collected basic soil properties and some flora surveys, and  investigated the mycorrhizal communities across the gradient using next generation sequencing. Soils will be collected at different elevations, corresponding to previously collected data. The following data will be collected:

* Basic soil properties (texture, pH, total C, total N, NH~4~^+^,NO~3~^-^) will be analysed.
* Labile C will be analysed and the labile C / NO~3~^-^ ratio calculated.
* 16S rRNA genes will be sequenced using the Illumina MiSeq to reveal differences in microbial communities at different elevations.
* qPCR will be performed to determine gene abundance of N cycling genes.
* A clone library for *nrfA* will be constructed, to investigate *nrfA* diversity and compare diversity with sequences previously identified as *nrfA* in the FUNGENE database
* Soil from sites with the highest and lowest C / NO~3~ ratio will be collected for a ^15^N tracing experiment, comparing N cycling rates at 60% and 100% water filled pore space (WFPS).

**All analyses apart from the ^15^N tracing experiment will be completed by 16 June, 2019.**

It is hypothesised that:

* Soils with the highest labile C / NO~3~ ratio will have the highest gene abundance of *nrfA*;
* Soils with the highest labile C / NO~3~ ratio will have the highest rate of DNRA.
* The DNRA rate will be higher at higher WFPS.
* Soil properties and plant species identity will both predict soil microbial community composition [@leff_predicting_2018].

#### Soil microbial communities and N cycling in agricultural soils and reforested agricultural soils

Soil samples will be collected from 5 contrasting paired agricultural soils and reforested sites nearby, as studied in [@england_previous_2016]. I have contacted the lead author of this paper and will obtain site and soil information to select 5 paired sites for study. Similar data will be collected at each site as in the previous experiment. I hypothesize that:

* Reforested soils will have greater gene abundance of *nrfA* than nearby agricultural soils.
* Reforested soils will have higher labile C / NO~3~ ratio.
* Reforested soils will have higher rates of DNRA.

#### Microbial diversity and DNRA rates in Australian rainforest and agricultural soils

I have identified potential field sites in Queensland, Victoria and Tasmania:

The "[TERN](https://supersites.tern.org.au/)" (Terrestrial Ecosystem
Research Sites) look ideal, because there is long term data for soil
moisture, climate, and lots of other research there:

https://supersites.tern.org.au/

##### Queensland
The sites in Far North Queensland [Daintree](https://supersites.tern.org.au/supersites/fnqr/daintree) and/or [Robson Creek](https://supersites.tern.org.au/supersites/fnqr/robson) look good. There's
no agriculture anywhere near the Daintree site, but it looks like there is
grass about 5 km away from the Robson Creek site just judging from what
I can see on Google Earth, at [17.121 S, 145.634 E](https://www.google.com.au/maps/place/17%C2%B007%2715.6%22S+145%C2%B038%2702.4%22E/@-17.121,145.6318113,693m/data=%213m2%211e3%214b1%214m5%213m4%211s0x0:0x0%218m2%213d-17.121%214d145.634).

##### Tasmania
I could contrast these with the site in Tasmanian site (called ["Warra Tall Eucalypt"](https://supersites.tern.org.au/supersites/wrra>).  [-43.09502 S, 146.65452 E](https://www.google.com.au/maps/place/17%C2%B007%2715.6%22S+145%C2%B038%2702.4%22E/@-17.121,145.6318113,693m/data=%213m2%211e3%214b1%214m5%213m4%211s0x0:0x0%218m2%213d-17.121%214d145.634),
It is mostly forest all around there, but it looks like there is some grass about 20 km
away.



## References
