---
title: "3rd Year Annual Review: Microbial communities performing DNRA in Australian forest soils"
author: "Eric Ireland"
date: "`r Sys.Date()`"
output: bookdown::word_document2
bibliography: references.bib
csl: asa-cssa-sssa.csl
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE)
knitr::opts_chunk$set(dpi=300,fig.width=7)
# library(plyr)
library(lattice)
library(readr)
library(ggplot2)
library(scales)
library(car)
library(tidyverse)
library(readxl)
library(knitr)
library(png)
library(readxl)
library(citr)
library(officer)
library(flextable)
# library(RColorBrewer)
# chapters <- read_csv("chapters.csv", col_types = cols(.default = "c"))
# chapters <- read_csv("chapters.csv", col_types = cols(`No. of papers expected from chapter` = col_character(), 
#     `Progress on lab/field task (%)` = col_character(), 
#     `Progress on writing task (%)` = col_character()))
# chapters <- read_csv("chapters.csv")
chapters <- read_csv("chapters.csv", col_types = cols(chap.no = col_integer(), 
     progress.lab = col_character (), 
     progress.writing = col_character()))

narG <- read_csv("narG.csv")
nrfA <- read_csv("nrfA.csv")
nosZ <- read_csv("nosZ.csv")
rRNA16S <- read_csv("bact16S.csv")
# rRNA16S <- read_csv("bact16S.csv", col_types = cols(sample_elevation = col_factor(levels = c("200","300", "500", "600", "800", "900", "1000", "1100", "1200", "1300", "1400", "1500"))))
nirK <- read_csv("nirK.csv")
```


```{r chapter_table,echo=FALSE}
chapter_table <- flextable(chapters)
chapter_table <- theme_box(chapter_table)
chapter_table <- bg(chapter_table, bg = "gray89", part = "header")
chapter_table <- merge_h_range(chapter_table, i = ~is.na(chap.no), j1 = "chap.description", j2 = "finish.date")
chapter_table <- bold(chapter_table, i = ~!is.na(chap.no))
chapter_table <- align( chapter_table, align = "center", part="header" )
center_key <- c("progress.lab","progress.writing")
chapter_table <- align( chapter_table, align = "center", j=center_key,part="body")
chapter_table <- font(chapter_table, part="all", fontname = "Times")
chapter_table <- fontsize(chapter_table, part="all", size = 10)
chapter_table <- set_header_labels(chapter_table, chap.no="Chap. No.",
                                   chap.description="Chapter description",
                                   progress.lab="Progress lab/field",
                                   progress.writing="Progress writing",
                                   finish.date="Planned finish date")
chapter_table <- height_all(chapter_table, height = .3)
chapter_table <- width(chapter_table, j="chap.description", width = 3.6)
chapter_table <- width(chapter_table, j="chap.no", width = .6)
big_border = fp_border(color="black", width = 2)
chapter_table <- hline(chapter_table, i = ~is.na(chap.no), border = big_border, part = "body" )
chapter_table <- hline(chapter_table,border = big_border,part = "header")
chapter_table <- set_caption(chapter_table, "Table 1. Thesis Chapters and Research Plan.") #can't cross reference tables in flextable 0.5.5, unfortunately. see https://stackoverflow.com/questions/56676952/table-cross-references-in-bookdown-with-ms-word-output
chapter_table
```

# Introduction

This report summarises progress since the second year annual review (in March 2018). Since that time, I have narrowed the scope of my PhD to forest soils only. The PhD will be written in the traditional format, instead of thesis by publication.

# Chapter 3
## Effect of N addition on gross nitrogen cycling and gene abundance in a temperate rainforest soil

The aim of this incubation experiment was to examine the effect of different C/N ratios on gross N cycling rates in forest soil, and the abundance of associated N-cycling genes in the microbial community. Soil was collected on from an area of cool temperate rainforest in Yarra State Forest near Powelltown in Victoria, Australia.

N was added in the form of NH~4~NO~3~ at a rate of 25 μg.g^−1^, 75 μg.g^−1^ and 150 μg.g^−1^ soil. Soil samples of 10 g were placed in 250 ml plastic vials and incubated at 25&deg;C and 60 % WFPS. Each sample had three replicates. Soils were destructively sampled after 15 min (for day 0 measurements), then at 7 d, 14 d and 28 d. The 16S rRNA gene, DNRA gene (*nrfA*) denitrification (*nirK*,*nirS*,*narG*,*nosZ*) and nitrification (AOA *amoA* and AOB *amoA*) genes were quantified using real-time PCR (qPCR).

### Work still to be completed:
1. Analyse ^15^NH~4~^+^ and ^15^NO~3~^-^ in KCl extracts.
2. Calculate gross N transformation rates (in collaboration with Prof. Christoph Müller).
3. Write thesis chapter.


# Chapter 4
## Soil microbial communities and microbial N-cycling gene abundance along an elevational gradient at Mount Baw Baw

Sabine Kasel and Craig Nitschke at Burnley had weather stations positioned along an elevational gradient at Mount Baw Baw. They have  collected soil properties and performed flora surveys, and investigated the mycorrhizal communities across the gradient using next generation sequencing. They also have 16S rRNA sequence data, which has not yet been analysed.

Soils were collected from 4 subplots at 12 sites at different elevations in April 2018, as shown in Figures \@ref(fig:dry) - \@ref(fig:alpine)). The following data were collected:

* Mineral N (NH~4~^+^,NO~3~^-^) in fresh soil
* Labile C was analysed and the labile C / NO~3~^-^ ratio calculated.
* 16S rRNA amplicons sequencing (metabarcoding) via Illumina MiSeq (at AGRF) to reveal differences in microbial communities at different elevations.
* Sequence data has been processed by Dr. Qinglin Chen to analyse OTUs and calculate alpha and beta diversity using QIIME 2 [@bolyenReproducibleInteractiveScalable2019].
* qPCR was performed to determine gene abundance of N cycling genes.
* Soil from sites with the contrasting C / NO~3~ ratios were  collected for a ^15^N tracing laboratory incubation experiment, comparing N cycling rates at 60% and 95% water holding capacity (WHC). Data from this experiment will be used for Chapter 5.


```{r dry, echo=FALSE, fig.cap="\\label{fig:dry}Dry sclerophyll forest (low elevation)."}
knitr::include_graphics("img/dry_sclerophyll.png")
```


```{r wet, echo=FALSE, fig.cap="\\label{fig:wet}Wet sclerophyll forest (mid elevation)."}
knitr::include_graphics("img/wet_sclerophyll.png")
```


```{r alpine, echo=FALSE, fig.cap="\\label{fig:alpine}Alpine vegetation (high elevation)."}
knitr::include_graphics("img/alpine.png")
```


### Work still to be completed:

* Basic soil properties (pH, total C, total N).
* Combined microbiome data with soil properties, mycorrhizal fungal data and vegetation and climate data in collaboration with Sabine Kasel and Craig Nitschke from Burnley.

### Baw Baw labile C and mineral N, April 2018

Labile C was measured as hot water-extractable carbon (HWC) using the method described by @ghaniHotwaterExtractableCarbon2003. Results are shown in Table 2.


``` {r TOC, echo=FALSE, message = FALSE}
bawbaw_TOC_50to59 <- read_table2(col_names = FALSE, "TOC_results/2018-07-02_Baw_Baw_50to59_diluted_1to20.txt", skip = 13)


newnames <- scan("50to59_sample_names", character(), quote = "") #replace "untitled" with sample names in 50to59
bawbaw_TOC_50to59$X3 <- newnames
bawbaw_TOC_50to59 %>%
  rename(`Sample name` = X3, `sample ID`=X4, `TOC mg/L` = X5, `TC mg/L` = X6, `IC mg/L` = X7) %>%
  select(`Sample name`:`IC mg/L`) %>%
  write_csv("TOC_results/2018-07-02_Baw_Baw_50to59_diluted_1to20_filtered_results.csv") # write raw results to csv file for printing in lab notebook

bawbaw_TOC_50to59$X3 <- gsub("m_SP.*", "", bawbaw_TOC_50to59$X3) # remove "_SP.*" from "summit" sample names to just give elevation.
bawbaw_TOC_50to59$X3 <- gsub("_SP.*", "", bawbaw_TOC_50to59$X3) # remove "_SP.*" from "summit" sample names to just give elevation.

bawbaw_TOC_60to97 <- read_table2(col_names = FALSE, "TOC_results/2018-07-04_Baw_Baw_60to97_diluted_1to20.txt", skip = 13)
bawbaw_TOC_60to97 %>%
  rename(`Sample name` = X3, `sample ID`=X4, `TOC mg/L` = X5, `TC mg/L` = X6, `IC mg/L` = X7) %>%
  select(`Sample name`:`IC mg/L`) %>%
  write_csv("TOC_results/2018-07-02_Baw_Baw_60to97_diluted_1to20_filtered_results.csv") # write raw results to csv file for printing in lab notebook


bawbaw_TOC_60to97 <- bawbaw_TOC_60to97[-c(4, 20), ] # delete 200SP2_... because they are duplicates (mislabelled?)
bawbaw_TOC_60to97$X3 <- gsub("m_SP.*", "", bawbaw_TOC_60to97$X3) # remove "_SP.*" from sample names to just give elevation.
bawbaw_TOC_60to97$X3 <- gsub("_SP.*", "", bawbaw_TOC_60to97$X3) # remove "_SP.*" from "summit" sample names to just give elevation.


bawbaw_TOC_means <- bind_rows(bawbaw_TOC_50to59, bawbaw_TOC_60to97) %>%
  rename(elevation = X3, TOC_mg_per_l = X5) %>%
  select(elevation, TOC_mg_per_l) %>%
  mutate(elevation=replace(elevation, elevation=="summit", "1500")) %>%
  mutate(TOC_mg_per_l_before_dilution = TOC_mg_per_l*20) %>% 
  mutate(mg_TOC_per_30ml = TOC_mg_per_l_before_dilution * 0.030) %>%
  mutate(ug_TOC_per_30ml = mg_TOC_per_30ml * 1000 ) %>% 
  mutate(ug_TOC_per_g_soil = ug_TOC_per_30ml / 3) %>% # calculate original concentration
  group_by(elevation) %>% 
  summarise(mean_ug_TOC_per_g_soil = mean(ug_TOC_per_g_soil))

# read mineral N data

bawbaw_mineral_N <- read_excel("18. 3422 Eric 180704 - SFA Report 18.07.10.xls", 
                                                       skip = 14) #read KCl data

bawbaw_mineral_N[bawbaw_mineral_N=="<0.2"] <- 0.2 # N.B.!! first need to convert <0.2 to 0.2
bawbaw_mineral_N$sample <- gsub("SP.", "", bawbaw_mineral_N$sample) # remove "SP." from sample names to just give elevation.

bawbaw_mineral_N_means <- bawbaw_mineral_N %>%
  rename(elevation = sample, nitrate = `N-Nitrate (mg/L)`, ammonium = `N-Ammonium (mg/L)`) %>% #rename column to make it easier to type!
  select(elevation, nitrate, ammonium) %>% #just keep the necessary data columns
  filter(elevation != "blank") %>% # remove blank samples
  mutate(nitrate = as.numeric(nitrate)) %>% # convert from character to numeric data
  mutate(ammonium = as.numeric(ammonium)) %>% # convert from character to numeric data
  mutate(elevation =replace(elevation, elevation=="summit", "1500")) %>% # replace "summit" with "1500"
  mutate(total_NO3_in_extract_mg = nitrate * 0.1,
             total_NO3_in_extract_ug = total_NO3_in_extract_mg * 1000,
             total_NO3_per_g_soil_ug = total_NO3_in_extract_ug / 10) %>% # calculate mean for nitrate
  mutate(total_NH4_in_extract_mg = ammonium * 0.1,
         total_NH4_in_extract_ug = total_NH4_in_extract_mg * 1000,
         total_NH4_per_g_soil_ug = total_NH4_in_extract_ug / 10) %>% # calculate mean for ammonium
  group_by(elevation) %>% 
  summarise_at(c("total_NO3_per_g_soil_ug", "total_NH4_per_g_soil_ug"), mean)


# calculate C/NO3- ratio and C/mineral-N ratio
combined_NC_means <- bind_cols(bawbaw_mineral_N_means, bawbaw_TOC_means)
combined_NC_means <- combined_NC_means %>% 
  select(-elevation1) %>% #remove extra elevation column
  mutate(C_nitrate_ratio = mean_ug_TOC_per_g_soil / total_NO3_per_g_soil_ug,
                                                  C_mineral_N_ratio = mean_ug_TOC_per_g_soil / (total_NO3_per_g_soil_ug + total_NH4_per_g_soil_ug))

# write .rds and .csv and .tsv file with data
write_rds(combined_NC_means,"combined_NC_means.rds")
write_csv(combined_NC_means,"combined_NC_means.csv")
write_tsv(combined_NC_means,"combined_NC_means.tsv")



#print see https://cran.r-project.org/web/packages/kableExtra/vignettes/awesome_table_in_html.html

combined_NC_means$C_nitrate_ratio <- round(combined_NC_means$C_nitrate_ratio, 1)
combined_NC_means$C_mineral_N_ratio <- round(combined_NC_means$C_mineral_N_ratio, 1)
combined_NC_means$mean_ug_TOC_per_g_soil <- round(combined_NC_means$mean_ug_TOC_per_g_soil, 1)
combined_NC_means <- combined_NC_means %>%
  mutate(elevation = factor(elevation, levels = c("200", "300", "500", "600", "800", "900", "1000", "1100","1200", "1300", "1400","1500"))) %>% 
  arrange(elevation) %>% 
  select(elevation, mean_ug_TOC_per_g_soil, total_NO3_per_g_soil_ug, total_NH4_per_g_soil_ug, C_nitrate_ratio) 
# names(combined_NC_means) <- c("Elevation", "mean HWC (µg/g soil)", "HWC/nitrate-N", "HWC/mineral-N")

NC_flextable <- flextable(combined_NC_means)
# NC_flextable <- set_header_labels(NC_flextable, elevation = "Elevation (m)", 
#     mean_ug_TOC_per_g_soil = "Mean HWC (µg/g soil)", Petal.Length = "Petal",
#     Petal.Width = "Petal" )

flextable_key <- c("mean_ug_TOC_per_g_soil", "total_NO3_per_g_soil_ug", "total_NH4_per_g_soil_ug","C_nitrate_ratio")
NC_flextable <- colformat_num(x = NC_flextable, col_keys = flextable_key, big.mark = "", digits=1)
NC_flextable <- compose(NC_flextable, part = "header", j = "elevation", 
    value = as_paragraph(as_b("Elevation (m)")))
NC_flextable <- compose(NC_flextable, part = "header", j = "mean_ug_TOC_per_g_soil", 
    value = as_paragraph(as_b("HWC (µg.g"), as_sup("-1"),as_b(".soil)")))
NC_flextable <- compose(NC_flextable, part = "header", j = "C_nitrate_ratio", 
    value = as_paragraph(as_b("HWC/"),as_b("NO"), as_sub("3"), as_sup("-")))
NC_flextable <- compose(NC_flextable, part = "header", j = "total_NH4_per_g_soil_ug", 
    value = as_paragraph(as_b("NH"), as_sub("4"), as_sup("+"),as_b("(µg.g"), as_sup("-1"), as_b(".soil)")))
NC_flextable <- compose(NC_flextable, part = "header", j = "total_NO3_per_g_soil_ug", 
    value = as_paragraph(as_b("NO"), as_sub("3"), as_sup("-"),as_b("(µg.g"), as_sup("-1"), as_b(".soil)")))
NC_flextable <- compose(NC_flextable, i = ~ total_NO3_per_g_soil_ug <2.1, j = "total_NO3_per_g_soil_ug",
  value = as_paragraph("<",
                       as_chunk(sprintf("%.01f",total_NO3_per_g_soil_ug)))) #
#                       ,as_chunk("*", props = fp_text(color = "red")))) # make a red asterix if <2.1
NC_flextable <- footnote( NC_flextable, i = ~ total_NO3_per_g_soil_ug <2.1, j = "total_NO3_per_g_soil_ug",
            value = as_paragraph("Below detection limit."),
            ref_symbols = "a")
NC_flextable <- align(NC_flextable, align = "right", part = "body")
NC_flextable <- autofit(NC_flextable)

NC_flextable <- set_caption(NC_flextable, "Table 2. Labile C as hot water extractable carbon (HWC) and mineral N (NH~4~^+^,NO~3~^-^) per gram soil.")

NC_flextable

#  kable_styling(bootstrap_options = "striped", full_width = F)
```


#### Baw Baw gene abundance, April 2018


<!-- ```{r rRNA16S, echo=FALSE, , message = FALSE, fig.cap="*rRNA16S* gene abundance in fresh Baw Baw soil, collected April 2018. Error bars show s. e. m. qPCR amplification efficiency = 100.2%, $R^{2} = 0.99$. \\label{rRNA16Sgraph}"} -->
<!-- rRNA16S$copiespergram <- (rRNA16S$SQ*1000)/0.25 #100ÂµL of solution is recovered from 0.25g soil. This template DNA is diluted 1:10 and then 1ÂµL of that diluted DNA is used as template. So we're counting copies of DNA in 0.1ÂµL of template. So, we are using 0.001 (1/100th) of the original DNA. So, times by 100 to get the amount in 100ÂµL DNA, then times by 4 (i.e. divide by 0.25) to get the copies in 1g of soil. -->
<!-- rRNA16S$log10copiespergram <- log10((rRNA16S$SQ*1000)/0.25) #log base 10 transform data -->
<!-- rRNA16S$logecopiespergram <- log((rRNA16S$SQ*1000)/0.25) #log base e transform data -->

<!-- # colnames(rRNA16S)[3] <- "time" # renamed day column to time -->



<!-- ### works!! -->

<!-- #summarySE function for errorbars from http://www.cookbook-r.com/Graphs/Plotting_means_and_error_bars_(ggplot2)/#Helper%20functions -->
<!-- summarySE <- function(data=NULL, measurevar, groupvars=NULL, na.rm=FALSE, -->
<!--                       conf.interval=.95, .drop=TRUE) { -->
<!--    -->

<!--   # New version of length which can handle NA's: if na.rm==T, don't count them -->
<!--   length2 <- function (x, na.rm=FALSE) { -->
<!--     if (na.rm) sum(!is.na(x)) -->
<!--     else       length(x) -->
<!--   } -->

<!--   # This does the summary. For each group's data frame, return a vector with -->
<!--   # N, mean, and sd -->
<!--   datac <- ddply(data, groupvars, .drop=.drop, -->
<!--                  .fun = function(xx, col) { -->
<!--                    c(N    = length2(xx[[col]], na.rm=na.rm), -->
<!--                      mean = mean   (xx[[col]], na.rm=na.rm), -->
<!--                      sd   = sd     (xx[[col]], na.rm=na.rm) -->
<!--                    ) -->
<!--                  }, -->
<!--                  measurevar -->
<!--   ) -->

<!--   # Rename the "mean" column     -->
<!--   datac <- rename(datac, c("mean" = measurevar)) -->

<!--   datac$se <- datac$sd / sqrt(datac$N)  # Calculate standard error of the mean -->

<!--   # Confidence interval multiplier for standard error -->
<!--   # Calculate t-statistic for confidence interval:  -->
<!--   # e.g., if conf.interval is .95, use .975 (above/below), and use df=N-1 -->
<!--   ciMult <- qt(conf.interval/2 + .5, datac$N-1) -->
<!--   datac$ci <- datac$se * ciMult -->

<!--   return(datac) -->
<!-- } -->

<!-- rRNA16S_summary <- summarySE(rRNA16S, measurevar="log10copiespergram", groupvars="sample_elevation") #not sure if this is correct! -->

<!-- rRNA16S2 <- rRNA16S_summary -->
<!-- rRNA16S2$sample_elevation <- factor(rRNA16S2$sample_elevation, levels=c("200","300","500","600","800","900","1000","1100","1200","1300","1400","1500")) -->
<!-- # rRNA16S2$time <- factor(rRNA16S2$time) -->


<!-- # Error bars represent standard error of the mean -->
<!-- rRNA_bars <- ggplot(rRNA16S2, aes(x=sample_elevation, y=log10copiespergram)) +  -->
<!--     geom_bar(position=position_dodge(), stat="identity",fill="steelblue") + -->
<!--     geom_errorbar(aes(ymin=log10copiespergram-se, ymax=log10copiespergram+se), -->
<!--                   width=.2,                    # Width of the error bars -->
<!--                   position=position_dodge(.9))+ -->
<!--   labs( -->
<!--     x = "sample elevation (m)", -->
<!--     y = expression("log"[10]~"(16S rRNA gene copies.g"^-1~" soil)") -->
<!--   )+ -->
<!--   scale_y_log10() -->

<!-- rRNA_bars + theme_classic() -->
<!-- ``` -->



#### Baw Baw gene abundance, April 2018


```{r rRNA16S, echo=FALSE, message = FALSE, fig.cap="*rRNA16S* gene abundance in fresh Baw Baw soil, collected April 2018. Error bars show s. e. m. qPCR amplification efficiency = 100.2%, $R^{2} = 0.99$. \\label{rRNA16Sgraph}"}
rRNA16S <- rRNA16S %>% mutate(rRNA16Spergram =  (SQ*1000)/0.25) %>%
  mutate_at(vars(sample_elevation), factor)

#100ÂµL of solution is recovered from 0.25g soil. This template DNA is diluted 1:10 and then 1ÂµL of that diluted DNA is used as template. So we're counting copies of DNA in 0.1ÂµL of template. So, we are using 0.001 (1/100th) of the original DNA. So, times by 100 to get the amount in 100ÂµL DNA, then times by 4 (i.e. divide by 0.25) to get the copies in 1g of soil.
# rRNA16S$log10copiespergram <- log10((rRNA16S$SQ*1000)/0.25) #log base 10 transform data
# rRNA16S$logecopiespergram <- log((rRNA16S$SQ*1000)/0.25) #log base e transform data

# colnames(rRNA16S)[3] <- "time" # renamed day column to time



### works!!

#summarySE function for errorbars from http://www.cookbook-r.com/Graphs/Plotting_means_and_error_bars_(ggplot2)/#Helper%20functions
# summarySE <- function(data=NULL, measurevar, groupvars=NULL, na.rm=FALSE,
#                       conf.interval=.95, .drop=TRUE) {
# 
#   # New version of length which can handle NA's: if na.rm==T, don't count them
#   length2 <- function (x, na.rm=FALSE) {
#     if (na.rm) sum(!is.na(x))
#     else       length(x)
#   }
# 
#   # This does the summary. For each group's data frame, return a vector with
#   # N, mean, and sd
#   datac <- ddply(data, groupvars, .drop=.drop,
#                  .fun = function(xx, col) {
#                    c(N    = length2(xx[[col]], na.rm=na.rm),
#                      mean = mean   (xx[[col]], na.rm=na.rm),
#                      sd   = sd     (xx[[col]], na.rm=na.rm)
#                    )
#                  },
#                  measurevar
#   )
# 
#   # Rename the "mean" column
#   datac <- rename(datac, c("mean" = measurevar))
# 
#   datac$se <- datac$sd / sqrt(datac$N)  # Calculate standard error of the mean
# 
#   # Confidence interval multiplier for standard error
#   # Calculate t-statistic for confidence interval:
#   # e.g., if conf.interval is .95, use .975 (above/below), and use df=N-1
#   ciMult <- qt(conf.interval/2 + .5, datac$N-1)
#   datac$ci <- datac$se * ciMult
# 
#   return(datac)
# }
# 
# rRNA16S_summary <- summarySE(rRNA16S, measurevar="log10copiespergram", groupvars="sample_elevation") #not sure if this is correct!
# 
# rRNA16S2 <- rRNA16S_summary
# rRNA16S2$sample_elevation <- factor(rRNA16S2$sample_elevation, levels=c("200","300","500","600","800","900","1000","1100","1200","1300","1400","1500"))
# # rRNA16S2$time <- factor(rRNA16S2$time)


# Error bars represent standard error of the mean

# Function to produce summary statistics (mean and +/- sd)

# rRNA16S$sample_elevation <- factor(rRNA16S2$sample_elevation, levels=c("200","300","500","600","800","900","1000","1100","1200","1300","1400","1500"))


rRNA_bars <- ggplot(rRNA16S, aes(x=sample_elevation, y=rRNA16Spergram)) +
    stat_summary(geom = "bar", fun.y = mean, position = "dodge") +
  stat_summary(geom = "errorbar", fun.data = mean_se, position = "dodge",width=.2) +
  labs(
    x = "sample elevation (m)",
    y = expression("16S rRNA gene copies.g"^-1~" soil")
  )+
  scale_y_log10(breaks = c(1, 100,10000,1000000,100000000,10000000000,10000000000000), limits = c(1, 100000000000))
  

rRNA_bars + theme_classic()
```

```{r rRNA_F-test}
rRNA16S$log10copiespergram <- log10((rRNA16S$SQ*1000)/0.25) #log base 10 transform data
rRNA16S$logecopiespergram <- log((rRNA16S$SQ*1000)/0.25) #log base e transform data
linear.model.rRNA <- lm(formula = log10copiespergram ~ sample_elevation, data = rRNA16S)
anovatablerRNA <- Anova(linear.model.rRNA, type="II")
rRNAp <- unname(anovatablerRNA$`Pr(>F)`[1])
```

16S rRNA gene abundance did not vary significantly between elevations (*F*-test, $p=`r ifelse(rRNAp < 0.01, "<0.01", ifelse(rRNAp < 0.05, "<0.05",round(rRNAp, 3)))`$).


# References
